hbase-server/src/test/java/org/apache/hadoop/hbase/MiniHBaseCluster.java

```diff
@@ -253,6 +256,12 @@ private void init(final int nMasterNodes, final int numAlwaysStandByMasters,
         hbaseCluster.addRegionServer(rsConf, i, user);
       }
 
+      // manually add the compaction servers as other users
+      for (int i = 0; i < numCompactionServers; i++) {
```

Comment:
Does compaction server have to be this integrated?

Reply:
Can it be moved outside hbase-server with its own test suite in its on hbase-compaction-server module? (Sorry, late to the game here)

Reply:
You mean the compaction server and test code move to an independent module hbase-compaction-server, and this module depend on hbase-server module. I agree it's necessary.
Maybe we need introduce a wrap class `MiniHBaseClusterWithExtraServer` in module hbase-server, it combine `MiniHBaseCluster` and `AbstractServer`(can be compaction server or replication server).And `MiniHBaseClusterWithExtraServer` can be used in hbase-compaction-server and hbase-replication-server module.

Reply:
@saintstack ping...

Reply:
Think twice, ITBLL need cover compaction server role, HBaseCluster has three types of cluster deployments: `MiniHBaseCluster`,`DistributedHBaseCluster`,`ProcessBasedLocalHBaseCluster`, all these classes should handle compaction server, maybe replication server also. So, It is better to integrate compaction server to `MiniHBaseCluster` for now? @saintstack